tests/test_logging.py

```diff
@@ -134,38 +129,42 @@ def test_configure_extended_logging(self):
         }
         opsdroid.configure_logging(config)
         rootlogger = logging.getLogger()
-        self.assertEqual(len(rootlogger.handlers), 1)
-        self.assertEqual(logging.StreamHandler, type(rootlogger.handlers[0]))
+        assert len(rootlogger.handlers) == 1
+        assert isinstance(rootlogger.handlers[0], logging.StreamHandler)
 
     def test_configure_default_logging(self):
         config = {}
         opsdroid.configure_logging(config)
         rootlogger = logging.getLogger()
-        self.assertEqual(len(rootlogger.handlers), 2)
-        self.assertEqual(logging.StreamHandler, type(rootlogger.handlers[0]))
-        self.assertEqual(rootlogger.handlers[0].level, logging.INFO)
-        self.assertEqual(
-            logging.handlers.RotatingFileHandler, type(rootlogger.handlers[1])
-        )
-        self.assertEqual(rootlogger.handlers[1].level, logging.INFO)
-        self.assertLogs("_LOGGER", "info")
-
-
-class TestWhiteAndBlackFilter(unittest.TestCase):
-    def setUp(self):
-        self.config = {
-            "logging": {
-                "path": False,
-                "level": "info",
-                "console": True,
-                "filter": {"whitelist": ["opsdroid"], "blacklist": ["opsdroid.core"]},
-            }
+        assert len(rootlogger.handlers), 2
+        # TODO use isinstance
+        assert isinstance(rootlogger.handlers[0], logging.StreamHandler)
+        assert rootlogger.handlers[0].level == logging.INFO
+        assert isinstance(rootlogger.handlers[1], logging.handlers.RotatingFileHandler)
+
+        assert rootlogger.handlers[1].level == logging.INFO
+        # self.assertLogs("_LOGGER", "info")
+
+
+@pytest.fixture
+def white_and_black_config():
+    # Set up
+    config = {
+        "logging": {
+            "path": False,
+            "level": "info",
+            "console": True,
+            "filter": {"whitelist": ["opsdroid"], "blacklist": ["opsdroid.core"]},
         }
+    }
+    yield config
+
+    # Tear down
+    config = {}
+    opsdroid.configure_logging(config)
 
-    def tearDown(self):
-        self.config = {}
-        opsdroid.configure_logging(self.config)
 
-    def test_configure_whitelist_and_blacklist(self):
-        opsdroid.configure_logging(self.config)
-        self.assertLogs("_LOGGER", "warning")
+class TestWhiteAndBlackFilter:
+    def test_configure_whitelist_and_blacklist(self, white_and_black_config):
+        opsdroid.configure_logging(white_and_black_config)
+        # self.assertLogs("_LOGGER", "warning")
```

Comment:
In here I think we could replace this assert with a caplog and assert if the warning was logged since whitelist and blacklist can't be used together.

Reply:
Hmm. 
I have tried using `caplog ` as described [here](https://docs.pytest.org/en/stable/logging.html#caplog-fixture). Or as @Waterbyte  used it [here](https://github.com/opsdroid/opsdroid/pull/1619/files). 
To my great surprise `caplog.text` is an empty string, even though this function call indeed creates logs at `Warning` level. All of the other properties of caplog (`messages`, `record_tuples` and `records`)  are empty lists, except `handler`.

See my code: 
```
def test_configure_whitelist_and_blacklist(self, caplog, white_and_black_config):
        caplog.set_level(logging.INFO)

        opsdroid.configure_logging(white_and_black_config)

        expected_log = '''Both whitelist and blacklist filters \
        found in configuration. Only one can be used at a time \
        - only the whitelist filter will be used.'''

        assert expected_log in caplog.text
```

I have tried setting the logger name to something else explicitly, like this:
> caplog.set_level(logging.INFO, logger='opsdroid.logging'),
None of them worked. 

Do you have a tip of guess why is this not working as it should?

Reply:
Ok, I resolved the problem with [capsys](https://docs.pytest.org/en/stable/capture.html#accessing-captured-output-from-a-test-function). I used `capsys` in the other two methods as well, where the test checks if there is no output at all as there is no logger configured.